src/librustc_mir/monomorphize/collector.rs

```diff
@@ -1113,14 +1124,16 @@ fn create_mono_items_for_default_impls<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
                         continue;
                     }
 
-                    if tcx.generics_of(method.def_id).own_counts().types != 0 {
+                    let counts = tcx.generics_of(method.def_id).own_counts();
+                    if counts.types + counts.consts != 0 {
```

Comment:
Shouldn't there be a method for this on `ty::Generics`?
If there is one but it recurses, you can make a version that doesn't and prefix its name with `own_`.